handler.py

```diff
@@ -0,0 +1,91 @@
+
+import os
+import json
+import urllib.parse
+
+from rdflib.plugins.sparql.parser import parseUpdate
+from rdflib.plugins.sparql.algebra import translateUpdate
+
+from sns_client import SnsClient
+from neptune_client import NeptuneClient
+
+
+def main(event, _):
+    rialto_sparql_endpoint = os.getenv('RIALTO_SPARQL_ENDPOINT', "http://localhost:8080/bigdata/namespace/kb/sparql")
+    rialto_sns_endpoint = os.getenv('RIALTO_SNS_ENDPOINT', "http://localhost:4575")
+    rialto_topic_arn = os.getenv('RIALTO_TOPIC_ARN', "rialto")
+    aws_region = os.getenv('AWS_REGION', "us-west-2")
+
+    sns_client = SnsClient(rialto_sns_endpoint, rialto_topic_arn, aws_region)
+    neptune_client = NeptuneClient(rialto_sparql_endpoint)
+
+    response, status_code = neptune_client.post(event['body'])
+
+    if status_code == 200:
+        if "update=" in event['body'] or event['Content-Type'] == "application/sparql-update":
+            entities = get_unique_subjects(
+                            get_entities(
+                                urllib.parse.unquote_plus(
+                                    event['body']).replace('update=', '')))
+            if entities:
+                message = {"Action": "touch", "Entities": entities}
+                _ = sns_client.publish(json.dumps(message))  # currently not using the neptune response
+
+    return {
+        'body': str(response),
+        'statusCode': status_code
+    }
+
+
+def get_entities(body):
+    delimeter_count = body.count("}};")  # determine if the sparql query is broken up by "}};"
+    if delimeter_count in [0, 1]:
+        return parse_body(body)
+
+    subjects = []
+    for chunk in body.split("}};"):
+        if len(chunk.rstrip()) > 0:
+            subjects += parse_body(chunk+"}};")  # append the "}};" that was removed by split
```

Comment:
Since `"}};"` is sprinkled around here, I might like to see this in a constant up above. (`STATEMENT_DELIMITER` or some such?) Not a big deal, though.